packages/main/src/components/time-zone/TimeZone.jsx

```diff
@@ -1,12 +1,58 @@
-import React from "react";
+import { React, useState, useEffect } from "react";
 import styles from "../time-zone/SettingsTab.module.css";
 import TimeZones from "../constants/TimeZone";
+import axios from "axios";
+
 const TimeZone = () => {
+  const [selectTimezone, setSelectTimezone] = useState([]);
+  const [user, setUser] = useState(null);
+
+  const handleSelect = e => {
+    setSelectTimezone(e.target.value);
+  };
+
+  let organization_id = JSON.parse(sessionStorage.organisations)[0].id;
+  let user_member_id = JSON.parse(sessionStorage.organisations)[0].member_id;
+
+  useEffect(() => {
+    const user = JSON.parse(sessionStorage.getItem("user"));
+    if (user) {
+      setUser(user);
+    }
+  }, []);
+
+  const handleTimezone = e => {
+    e.preventDefault();
+
+    axios
+      .patch(
+        `https://api.zuri.chat/organizations/${organization_id}/members/${user_member_id}/settings`,
+
+        {
+          languages_and_regions: {
+            time_zone: selectTimezone
+          }
+        },
+        {
+          headers: {
+            Authorization: "Bearer " + user.token
+          }
+        }
+      )
+      .then(res => {
+        const userTimeZone = JSON.parse(sessionStorage.getItem("user"));
+
+        userTimeZone.time_zone = selectTimezone;
+
+        sessionStorage.setItem("user", JSON.stringify(userTimeZone));
```

Comment:
You can also add a toast notification to signify that the change has successfully been made